MusicNotationKit/MusicNotationKit/NoteCollection.swift

```diff
@@ -25,6 +25,9 @@ public protocol NoteCollection {
      */
     var noteTimingCount: Int { get }
 
+    var first: Note { get }
```

Comment:
I could do that. I actually considered having this implement the Swift protocol that defines `first` and `last`, but that had way too much to it.

My reason for not wanting to make it optional is that it really isn't optional. It should never happen that `first` or `last` is optional, because you can not make a note less or more than a count of 1 and a Tuplet can only be instantiated with a valid set of notes, which must be greater than or equal to 1.

What do you think? I could make it optional just for safety, but for this, I feel like I could make the exception for throwing a `fatalError` since it's so impossible.